src/librustc_mir/interpret/operand.rs

```diff
@@ -541,7 +544,6 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
         // Early-return cases.
         match val.val {
```

Comment:
Could you add a doc comment to this function that say something similar to `resolve`, namely that `val` (and `layout`) is expected to be already in the `InterpCx` "universe"?